Service/EmailService.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PaneeDesign\DatabaseSwiftMailerBundle\Service;
+
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityNotFoundException;
+use Exception;
+use PaneeDesign\DatabaseSwiftMailerBundle\Entity\Email;
+use PaneeDesign\DatabaseSwiftMailerBundle\Repository\EmailRepository;
+use Swift_Mime_SimpleMessage;
+use Swift_SwiftException;
+
+class EmailService implements EmailServiceInterface
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    private $manager;
+
+    /**
+     * @var EmailRepository
+     */
+    private $repository;
+
+    public function __construct(EntityManagerInterface $manager, EmailRepository $repository)
+    {
+        $this->manager = $manager;
+        $this->repository = $repository;
+    }
+
+    /**
+     * @throws EntityNotFoundException
+     */
+    public function getById(int $id): Email
+    {
+        /* @var Email $email */
+        $email = $this->repository->find($id);
+
+        if (!$email) {
+            throw new EntityNotFoundException('Unable to find Email entity.');
+        }
+
+        return $email;
+    }
+
+    /**
+     * @param ?int $limit
+     * @param ?int $offset
+     *
+     * @return Email[]
+     */
+    public function paginate(?int $limit = null, ?int $offset = null): array
```

Comment:
"paginate" is not meaningful name for a method of this kind. The goal of this method is to return items: rename to something better like getItems(), getRows, getEmails